src/sentry/services/hybrid_cloud/app/serial.py

```diff
@@ -40,18 +42,36 @@ def serialize_sentry_app(app: SentryApp) -> RpcSentryApp:
     )
 
 
+def serialize_sentry_app_component(app_component: SentryAppComponent) -> RpcSentryAppComponent:
+    return RpcSentryAppComponent(
+        uuid=str(app_component.uuid),
+        sentry_app_id=app_component.sentry_app_id,
+        type=app_component.type,
+        app_schema=app_component.schema,
+    )
+
+
 def serialize_sentry_app_installation(
-    installation: SentryAppInstallation, app: Optional[SentryApp] = None
+    installation: SentryAppInstallation,
+    app: Optional[SentryApp] = None,
+    prepare_sentry_app_components_type: Optional[str] = None,
 ) -> RpcSentryAppInstallation:
     if app is None:
         app = installation.sentry_app
         assert app is not None
 
+    app_component = None
+    if prepare_sentry_app_components_type is not None:
+        app_component = installation.prepare_sentry_app_components(
+            prepare_sentry_app_components_type
+        )
```

Comment:
Since there are multiple component types. Should we have a method on the SentryAppInstallation that lets the receiver define what kind of component they are expecting? Right now one could load an installation with component a, but the logic requires component b and we have no way to prevent or detect that.